homeassistant/components/zwave_js/discovery.py

```diff
@@ -0,0 +1,131 @@
+        # NOTE: we do not use the value_id here because list values get a different value id
+        return f"{self.node.node_id}.{self.primary_value.property_}"
+
+
+@dataclass
+class ZWaveDiscoverySchema:
+    """Z-Wave discovery schema.
+
+    The (primary) value for an entity must match these conditions.
+    """
+
+    # specify the hass platform for which this scheme applies (e.g. light, sensor)
+    platform: str
+    # [optional] hint for platform
+    hint: Optional[str]
+    # [optional] the node's basic device class must match ANY of these values
+    device_class_basic: List[str] = field(default_factory=list)
+    # [optional] the node's generic device class must match ANY of these values
+    device_class_generic: List[str] = field(default_factory=list)
+    # [optional] the node's specific device class must match ANY of these values
+    device_class_specific: List[str] = field(default_factory=list)
+    # [optional] the value's command class must match ANY of these values
+    command_class: List[int] = field(default_factory=list)
+    # [optional] the value's endpoint must match ANY of these values
+    endpoint: List[int] = field(default_factory=list)
+    # [optional] the value's property must match ANY of these values
+    property: List[str] = field(default_factory=list)
+    # [optional] the value's metadata_type must match ANY of these values
+    type: List[str] = field(default_factory=list)
+
+
+DISCOVERY_SCHEMAS = [
+    # generic text sensors
+    ZWaveDiscoverySchema(
+        platform="sensor",
+        hint="string_sensor",
+        command_class=[
+            CommandClass.ALARM,
+            CommandClass.SENSOR_ALARM,
+            CommandClass.INDICATOR,
+            CommandClass.NOTIFICATION,
+        ],
+        type=["string"],
+    ),
+    # generic numeric sensors
+    ZWaveDiscoverySchema(
+        platform="sensor",
+        hint="numeric_sensor",
+        command_class=[
+            CommandClass.SENSOR_MULTILEVEL,
+            CommandClass.METER,
+            CommandClass.ALARM,
+            CommandClass.SENSOR_ALARM,
+            CommandClass.INDICATOR,
+            CommandClass.BATTERY,
+            CommandClass.NOTIFICATION,
+            CommandClass.BASIC,
+        ],
+        type=["number"],
+    ),
+]
+
+
+async def async_discover_values(node: ZwaveNode) -> AsyncGenerator[ZwaveDiscoveryInfo, None]:
+    """Run discovery on ZWave node and return matching (primary) values."""
+    for value in node.values.values():
+        disc_val = await async_discover_value(value)
+        if disc_val:
+            yield disc_val
+
+
+async def async_discover_value(value: ZwaveValue) -> Optional[ZwaveDiscoveryInfo]:
+    """Run discovery on Z-Wave value and return ZwaveDiscoveryInfo if match found."""
+    for schema in DISCOVERY_SCHEMAS:
+        # check device_class_basic
+        if not compare_value(schema.device_class_basic, value.node.device_class.basic):
```

Comment:
This is comparing two strings that are enums, right? Shouldn't we use a strict equality test instead of membership test?

Reply:
No, we’re comparing a value in a list here. See discovery schema.

Reply:
I've corrected the type hint.